### PR TITLE
*: refactor transaction documents

### PR DESCRIPTION
**Pull request by dcalvin**
### What is changed, added or deleted? 

Based on our reviews and offline discussions, we decided to refactor and update transaction-related documents, including:

- Split and merge the current Optimistic Locking best practice doc into existing Transaction Documents (because it is not a best practice doc, rather, it is a compilation of existing content on transactions.)
- The Transaction Overview doc include content applicable to both Optimistic Transaction and Pessimistic Transaction
- The previous Transaction Model is now changed to Optimistic Transaction 

@anotherrachel  FYI.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [x] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: v3.1/reference/transactions/overview.md [hunk 1]
```diff
@@ -160,3 +160,48 @@ rollback;
...
+* 最终导致事务完成提交的耗时增加。
+
+因此，TiDB 特意对事务的大小做了一些限制：
+
+* 单个事务包含的 SQL 语句不超过 5000 条（默认）
+* 每个键值对不超过 6 MB
+* 键值对的总数不超过 300000
+* 键值对的总大小不超过 100 MB
+
+为了使性能达到最优，建议每 100～500 行写入一个事务。
```
Comment (MyonKeminta) [inline v3.1/reference/transactions/overview.md]: @AndreMouche  @coocood Should this content be updated for 4.0?

**Comment by lilin90** [comment]
@dcalvin Please resolve lint failure.

**Comment by dcalvin** [comment]
> @dcalvin Please resolve lint failure.
@anotherrachel will follow this PR through.

File: v3.1/reference/transactions/overview.md [hunk 2]
```diff
@@ -5,9 +5,9 @@ category: reference
 
 # TiDB 事务概览
 
-TiDB 支持完整的分布式事务。本文主要介绍涉及到事务的语句、显式/隐式事务以及事务的隔离级别和惰性检查。
+TiDB 支持完整的分布式事务，默认使用[乐观事务模型](/v3.1/reference/transactions/transaction-optimistic/)。事务提交时假定不会发生并发冲突，只有在事务最终提交时才会检测冲突。同时，TiDB V3.0 版本也加入了[悲观事务](v3.1/reference/transactions/transaction-pessimistic.md)的支持。本文主要介绍涉及到事务的语句、显式/隐式事务以及事务的隔离级别和惰性检查，以及事务大小的限制。
```
Comment (lilin90) [inline v3.1/reference/transactions/overview.md]: @anotherrachel Please confirm this description about default transaction model with @AndreMouche, to check whether the optimistic model is the default one in all versions.

File: v3.1/reference/transactions/overview.md [hunk 3]
```diff
@@ -160,3 +160,48 @@ rollback;
...
+
+## 事务大小
+
+对于 TiDB 事务而言，事务太大或者太小，都会影响事务性能。为了克服上述事务在处理过程中的不足，在实际应用中可以根据事务大小进行针对性处理。
+
+### 小事务
+
+在自动提交状态 (`autocommit = 1`) 下，下面三条语句各为一个事务：
+
+```sql
```
Comment (lilin90) [inline v3.1/reference/transactions/overview.md]: - Suggest moving the code comment out of the block.
- Add code snippets to make code copyable.

Please also update other code blocks in this PR.

File: v3.1/reference/transactions/overview.md (hunk 1 above)
Comment (lilin90) [reply v3.1/reference/transactions/overview.md]: @AndreMouche @coocood PTAL~ Thanks!

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
+---
+title: 乐观事务模型
+category: reference
+aliases: ['/docs-cn/v3.1/reference/transactions/transaction-model/']
+---
+
+# 乐观事务模型
+
+TiDB 默认使用乐观事务模型。也就是说，在执行 `UPDATE`、`INSERT`、`DELETE` 等语句时，只有在提交过程中才会检查写写冲突，而不是像 MySQL 一样使用行锁来避免写写冲突。类似的，诸如 `GET_LOCK()` 和 `RELEASE_LOCK()` 等函数以及 `SELECT .. FOR UPDATE` 之类的语句在 TiDB 和 MySQL 中的执行方式并不相同。所以业务端在执行 SQL 语句后，需要注意检查 `COMMIT` 的返回值，即使执行时没有出错，`COMMIT` 的时候也可能会出错。
```
Comment (lilin90) [inline v3.1/reference/transactions/transaction-optimistic.md]: Just to confirm: is "TiDB 默认使用乐观事务模型" for all versions?

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+![TiDB 中的两阶段提交](/media/2pc-in-tidb.png)
+
+1. 客户端开始一个事务。
+
+    TiDB 从 PD 获取一个全局唯一递增的版本号作为当前事务的开始版本号，这里定义为该事务的 `start_ts` 版本。
+
+2. 客户端发起读请求。
+
+    a. TiDB 从 PD 获取数据路由信息，即数据具体存在哪个 TiKV 节点上。
```
Comment (lilin90) [inline v3.1/reference/transactions/transaction-optimistic.md]:
```suggestion
    1. TiDB 从 PD 获取数据路由信息，即数据具体存在哪个 TiKV 节点上。
```

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+    b. TiDB 从 TiKV 获取 `start_ts` 版本下对应的数据信息。
```
Comment (lilin90) [inline v3.1/reference/transactions/transaction-optimistic.md]:
```suggestion
    2. TiDB 从 TiKV 获取 `start_ts` 版本下对应的数据信息。
```

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+3. 客户端发起写请求。
+
+    TiDB 校验写入数据是否符合一致性约束（如数据类型是否正确、是否符合唯一索引约束等）。**校验通过的数据将存放在内存里。**
+
+4. 客户端发起 commit。
+
+5. TiDB 开始两阶段提交，保证分布式事务的原子性，让数据真正落盘。
+
+    a. TiDB 从当前要写入的数据中选择一个 Key 作为当前事务的 Primary Key。
```
Comment (lilin90) [inline v3.1/reference/transactions/transaction-optimistic.md]: How about using 1, 2, 3, ...?

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+推荐通过以下两种方式进行参数设置：
+
+1. Session 级别设置：
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    set @@tidb_disable_txn_auto_retry = off;
```
Comment (lilin90) [inline v3.1/reference/transactions/transaction-optimistic.md]: Update this block to make it copy-friendly. Each SQL statement in one code block.

File: v3.1/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 9]
```diff
@@ -312,7 +312,7 @@ set @@global.tidb_distsql_scan_concurrency = 10;
 
 默认值：10
 
-这个变量用来设置最多可重试次数，即在一个事务执行中遇到可重试的错误(例如事务冲突、事务提交过慢或表结构变更)时，这个事务可以被重新执行，这个变量值表明最多可重试的次数。
+这个变量用来设置最大重试次数。一个事务执行中遇到可重试的错误(例如事务冲突、事务提交过慢或表结构变更)时，会根据该变量的设置进行重试。需注意 `tidb_retry_limit` 的值不能为 0，否则会**完全禁用自动重试**。
```
Comment (youjiali1995) [inline v3.1/reference/configuration/tidb-server/tidb-specific-variables.md]: 可以是 0，只是效果和关闭重试一样。

File: v3.1/reference/transactions/overview.md [hunk 10]
```diff
@@ -5,11 +5,11 @@ category: reference
 
 # TiDB 事务概览
 
-TiDB 支持完整的分布式事务。本文主要介绍涉及到事务的语句、显式/隐式事务以及事务的隔离级别和惰性检查。
+TiDB 支持完整的分布式事务，默认使用[乐观事务模型](/v3.1/reference/transactions/transaction-optimistic.md)。事务提交时假定不会发生并发冲突，只有在事务最终提交时才会检测冲突。同时，TiDB v3.0 版本也加入了[悲观事务](/v3.1/reference/transactions/transaction-pessimistic.md)的支持。本文主要介绍涉及到事务的语句、显式/隐式事务、事务的隔离级别和惰性检查，以及事务大小的限制。
```
Comment (youjiali1995) [inline v3.1/reference/transactions/overview.md]: 在 v3.0.8 之后默认使用悲观事务了。

File: v3.1/reference/transactions/transaction-optimistic.md (hunk 4 above)
Comment (youjiali1995) [reply v3.1/reference/transactions/transaction-optimistic.md]: v3.0.8 及以后默认使用悲观事务了。

File: v3.1/reference/mysql-compatibility.md [hunk 11]
```diff
@@ -13,7 +13,7 @@ TiDB 支持 MySQL 传输协议及其绝大多数的语法。这意味着您现
 
 > **注意：**
 >
-> 本页内容仅涉及 MySQL 与 TiDB 的总体差异。关于[安全特性](/v3.1/reference/security/compatibility.md)及[事务模型](/v3.1/reference/transactions/transaction-model.md)的兼容信息请查看各自具体页面。
+> 本页内容仅涉及 MySQL 与 TiDB 的总体差异。关于[安全特性](/v3.1/reference/security/compatibility.md)及[事务模型](/v3.1/reference/transactions/transaction-optimistic.md)的兼容信息请查看各自具体页面。
```
Comment (youjiali1995) [inline v3.1/reference/mysql-compatibility.md]: 还有悲观事务的差异：https://github.com/pingcap/docs-cn/blob/master/v3.1/reference/transactions/transaction-pessimistic.md#%E5%92%8C-mysql-innodb-%E7%9A%84%E5%B7%AE%E5%BC%82

File: v3.1/reference/sql/statements/select.md [hunk 12]
```diff
@@ -74,7 +74,7 @@ category: reference
 |`HAVING where_condition` | Having 子句与 Where 子句作用类似，Having 子句可以让过滤 GroupBy 后的各种数据，Where 子句用于在聚合前过滤记录。|
 |`ORDER BY` | OrderBy 子句用于指定结果排序顺序，可以按照列、表达式或者是 `select_expr` 列表中某个位置的字段进行排序。|
 |`LIMIT` | Limit 子句用于限制结果条数。Limit 接受一个或两个数字参数，如果只有一个参数，那么表示返回数据的最大行数；如果是两个参数，那么第一个参数表示返回数据的第一行的偏移量（第一行数据的偏移量是 0），第二个参数指定返回数据的最大条目数。|
-|`FOR UPDATE` | 对查询结果集所有行上锁（对于在查询条件内，但是不在结果集的行，将不会加锁，如事务启动后由其他事务写入的行），以监测其他事务对这些的并发修改。TiDB 使用[乐观事务模型](/v3.1/reference/transactions/transaction-model.md#事务模型)在语句执行期间不会检测锁，因此，不会像 PostgreSQL 之类的数据库一样，在当前事务结束前阻止其他事务执行 UPDATE、DELETE 和 SELECT FOR UPDATE。在事务的提交阶段 SELECT FOR UPDATE 读到的行，也会进行两阶段提交，因此，它们也可以参与事务冲突检测。如发生写入冲突，那么包含 SELECT FOR UPDATE 语句的事务会提交失败。如果没有冲突，事务将成功提交，当提交结束时，这些被加锁的行，会产生一个新版本，可以让其他尚未提交的事务，在将来提交时发现写入冲突。|
+|`FOR UPDATE` | 对查询结果集所有行上锁（对于在查询条件内，但是不在结果集的行，将不会加锁，如事务启动后由其他事务写入的行），以监测其他事务对这些的并发修改。TiDB 使用[乐观事务模型](/v3.1/reference/transactions/transaction-optimistic.md)在语句执行期间不会检测锁，因此，不会像 PostgreSQL 之类的数据库一样，在当前事务结束前阻止其他事务执行 UPDATE、DELETE 和 SELECT FOR UPDATE。在事务的提交阶段 SELECT FOR UPDATE 读到的行，也会进行两阶段提交，因此，它们也可以参与事务冲突检测。如发生写入冲突，那么包含 SELECT FOR UPDATE 语句的事务会提交失败。如果没有冲突，事务将成功提交，当提交结束时，这些被加锁的行，会产生一个新版本，可以让其他尚未提交的事务，在将来提交时发现写入冲突。|
```
Comment (youjiali1995) [inline v3.1/reference/sql/statements/select.md]: 若使用悲观事务，则行为与其他数据库相同，差异见 https://github.com/pingcap/docs-cn/blob/master/v3.1/reference/transactions/transaction-pessimistic.md#%E5%92%8C-mysql-innodb-%E7%9A%84%E5%B7%AE%E5%BC%82

File: v3.1/reference/transactions/transaction-optimistic.md (hunk 4 above)
Comment (sticnarf) [reply v3.1/reference/transactions/transaction-optimistic.md]: It's not totally right now. From TiDB 3.0.8 and 3.1.0-beta.1, pessimistic transactions are used by default. But clusters upgraded from older versions will keep using optimistic transactions.

Comment (youjiali1995) [reply v3.1/reference/transactions/transaction-optimistic.md]: 哦对，升级后还是一样的。

File: v3.1/reference/transactions/overview.md (hunk 1 above)
Comment (youjiali1995) [reply v3.1/reference/transactions/overview.md]: 需要更新，但大事务目前没有文档。我觉得可以等后面整理大事务文档时再更新。

Comment (sticnarf) [reply v3.1/reference/transactions/overview.md]: 这里还是 3.1 的文档，应该不用加入。

File: v3.1/reference/configuration/tidb-server/tidb-specific-variables.md (hunk 9 above)
Comment (anotherrachel) [reply v3.1/reference/configuration/tidb-server/tidb-specific-variables.md]: @youjiali1995 这样改如何？⬇️
```suggestion
这个变量用来设置最大重试次数。一个事务执行中遇到可重试的错误(例如事务冲突、事务提交过慢或表结构变更)时，会根据该变量的设置进行重试。需注意开启重试后， `tidb_retry_limit` 的值不能为 0，否则会**完全禁用自动重试**。
```

File: v3.1/reference/sql/statements/select.md (hunk 12 above)
Comment (anotherrachel) [reply v3.1/reference/sql/statements/select.md]: @youjiali1995 了解。我会补充进去

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+
+# TiDB 乐观事务模型
+
+TiDB 默认使用乐观事务模型，即事务提交时假定不会发生并发冲突，只有在事务最终提交时才会检测冲突，而不是像 MySQL 一样使用行锁来避免写写冲突。
+
+本文介绍 TiDB 乐观事务的原理，以及相关特性。本文假定你对 [TiDB 的整体架构](/v3.1/architecture.md#tidb-整体架构)、[Percolator](https://www.usenix.org/legacy/event/osdi10/tech/full_papers/Peng.pdf) 事务模型以及事务的 [ACID 特性](/v3.1/glossary.md#acid)都有一定了解。
+
+> **注意：**
+>
+> 目前，TiDB 3.0.8 默认使用悲观事务模型。
```
Comment (anotherrachel) [inline v3.1/reference/transactions/transaction-optimistic.md]:
```suggestion
> 自 TiDB 3.0.8 开始，默认使用悲观事务模型。
```

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 14]
```diff
@@ -1,15 +1,20 @@
...
+summary: 了解 TiDB 的乐观事务模型。
 category: reference
 aliases: ['/docs-cn/v3.1/reference/transactions/transaction-model/']
 ---
 
-# 乐观事务模型
+# TiDB 乐观事务模型
 
-TiDB 默认使用乐观事务模型。也就是说，在执行 `UPDATE`、`INSERT`、`DELETE` 等语句时，只有在提交过程中才会检查写写冲突，而不是像 MySQL 一样使用行锁来避免写写冲突。类似的，诸如 `GET_LOCK()` 和 `RELEASE_LOCK()` 等函数以及 `SELECT .. FOR UPDATE` 之类的语句在 TiDB 和 MySQL 中的执行方式并不相同。所以业务端在执行 SQL 语句后，需要注意检查 `COMMIT` 的返回值，即使执行时没有出错，`COMMIT` 的时候也可能会出错。
+TiDB 默认使用乐观事务模型，即事务提交时假定不会发生并发冲突，只有在事务最终提交时才会检测冲突，而不是像 MySQL 一样使用行锁来避免写写冲突。
```
Comment (sticnarf) [inline v3.1/reference/transactions/transaction-optimistic.md]: 是不是说清楚 < 3.0.7 的版本才是默认乐观事务模型。另外在注意那里提示，升级到 >= 3.0.8 的版本不改变默认事务模型。

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+    TiDB 校验写入数据是否符合一致性约束（如数据类型是否正确、是否符合唯一索引约束等）。**校验通过的数据将存放在内存里。**
+
+4. 客户端发起 commit。
+
+5. TiDB 开始两阶段提交，保证分布式事务的原子性，让数据真正落盘。
+
+    1. TiDB 从当前要写入的数据中选择一个 Key 作为当前事务的 Primary Key。
+    2. TiDB 从 PD 获取所有数据的写入路由信息，并将所有的 Key 按照所有的路由进行分类。
+    3. TiDB 并发地向所有涉及的 TiKV 发起 prewrite 请求。TiKV 收到 prewrite 数据后，检查数据版本信息是否存在冲突或已过期。符合条件的数据会被加锁。
+    4. TiDB 成功收到所有 prewrite 请求。
```
Comment (sticnarf) [inline v3.1/reference/transactions/transaction-optimistic.md]:
```suggestion
    4. TiDB 收到所有 prewrite 响应且所有 prewrite 都成功。
```

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+通过分析 TiDB 中事务的处理流程，可以发现 TiDB 事务有如下优点：
+
+* 实现原理简单，易于理解。
+* 基于单实例事务实现了跨节点事务。
+* 锁管理实现了去中心化。
+
+但 TiDB 事务也存在以下缺点：
+
+* 两阶段提交使网络交互增多。
+* 缺少一个中心化的版本管理服务。
```
Comment (sticnarf) [inline v3.1/reference/transactions/transaction-optimistic.md]: [原文](https://pingcap.com/blog-cn/best-practice-optimistic-transaction/) 的描述是正确的。

File: v3.1/reference/configuration/tidb-server/tidb-specific-variables.md (hunk 9 above)
Comment (youjiali1995) [reply v3.1/reference/configuration/tidb-server/tidb-specific-variables.md]: emmm，感觉也不太好。重试这个功能相当于有 2 个开关，是预期行为。。

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+
+## 事务的重试
+
+使用乐观事务模型时，在高冲突率的场景中，事务很容易提交失败。而 MySQL 内部使用的是悲观事务模型，在执行 SQL 语句的过程中进行冲突检测，所以提交时很难出现异常。为了兼容 MySQL 的悲观事务行为，TiDB 提供了重试机制。
+
+### 重试机制
+
+当事务提交后，如果发现冲突，TiDB 内部重新执行包含写操作的 SQL 语句。你可以通过设置 `tidb_disable_txn_auto_retry = off` 开启自动重试，并通过 `tidb_retry_limit` 设置重试次数：
+
+```toml
```
Comment (sticnarf) [inline v3.1/reference/transactions/transaction-optimistic.md]: 这里不是 toml，用 toml 的语法高亮导致了 `off` 颜色异常。
可以用 sql

File: v3.1/reference/configuration/tidb-server/tidb-specific-variables.md (hunk 9 above)
Comment (anotherrachel) [reply v3.1/reference/configuration/tidb-server/tidb-specific-variables.md]: @youjiali1995 How about this ⬇️
```suggestion
这个变量用来设置最大重试次数。一个事务执行中遇到可重试的错误(例如事务冲突、事务提交过慢或表结构变更)时，会根据该变量的设置进行重试。注意当 `tidb_retry_limit = 0` 时，也会禁用自动重试。
```

Comment (youjiali1995) [reply v3.1/reference/configuration/tidb-server/tidb-specific-variables.md]: LGTM

File: v3.1/reference/transactions/transaction-optimistic.md [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+summary: 了解 TiDB 的乐观事务模型。
+category: reference
+aliases: ['/docs-cn/v3.1/reference/transactions/transaction-model/']
+---
+
+# TiDB 乐观事务模型
+
+TiDB 默认使用乐观事务模型，即事务提交时假定不会发生并发冲突，只有在事务最终提交时才会检测冲突，而不是像 MySQL 一样使用行锁来避免写写冲突。
+
+本文介绍 TiDB 乐观事务的原理，以及相关特性。本文假定你对 [TiDB 的整体架构](/v3.1/architecture.md#tidb-整体架构)、[Percolator](https://www.usenix.org/legacy/event/osdi10/tech/full_papers/Peng.pdf) 事务模型以及事务的 [ACID 特性](/v3.1/glossary.md#acid)都有一定了解。
```
Comment (lilin90) [inline v3.1/reference/transactions/transaction-optimistic.md]: Suggest moving this para to the beginning of the body part. Keeping the current 1st para closer to the note would be more friendly.

Comment (anotherrachel) [reply v3.1/reference/transactions/transaction-optimistic.md]: @lilin90 Updated in [d2f7b5d](https://github.com/pingcap/docs-cn/pull/2198/commits/d2f7b5deb4c7783eddd666ee713029266b9b8231).

**Comment by anotherrachel** [comment]
@lilin90 PTAL again, thanks!

File: v3.1/reference/mysql-compatibility.md [hunk 19]
```diff
@@ -13,7 +13,7 @@ TiDB 支持 MySQL 传输协议及其绝大多数的语法。这意味着您现
 
 > **注意：**
 >
-> 本页内容仅涉及 MySQL 与 TiDB 的总体差异。关于[安全特性](/v3.1/reference/security/compatibility.md)及[事务模型](/v3.1/reference/transactions/transaction-model.md)的兼容信息请查看各自具体页面。
+> 本页内容仅涉及 MySQL 与 TiDB 的总体差异。关于[安全特性](/v3.1/reference/security/compatibility.md)、[悲观事务模型](/v3.1/reference/transactions/transaction-pessimistic.md#和-mysql-innodb的差异)的兼容信息请查看各自具体页面。
```
Comment (lilin90) [inline v3.1/reference/mysql-compatibility.md]:
```suggestion
> 本页内容仅涉及 MySQL 与 TiDB 的总体差异。关于[安全特性](/v3.1/reference/security/compatibility.md)、[悲观事务模型](/v3.1/reference/transactions/transaction-pessimistic.md#和-mysql-innodb-的差异)的兼容信息请查看各自具体页面。
```

File: v3.1/reference/sql/statements/select.md [hunk 20]
```diff
@@ -74,7 +74,7 @@ category: reference
...
 |`ORDER BY` | OrderBy 子句用于指定结果排序顺序，可以按照列、表达式或者是 `select_expr` 列表中某个位置的字段进行排序。|
 |`LIMIT` | Limit 子句用于限制结果条数。Limit 接受一个或两个数字参数，如果只有一个参数，那么表示返回数据的最大行数；如果是两个参数，那么第一个参数表示返回数据的第一行的偏移量（第一行数据的偏移量是 0），第二个参数指定返回数据的最大条目数。|
-|`FOR UPDATE` | 对查询结果集所有行上锁（对于在查询条件内，但是不在结果集的行，将不会加锁，如事务启动后由其他事务写入的行），以监测其他事务对这些的并发修改。TiDB 使用[乐观事务模型](/v3.1/reference/transactions/transaction-model.md#事务模型)在语句执行期间不会检测锁，因此，不会像 PostgreSQL 之类的数据库一样，在当前事务结束前阻止其他事务执行 UPDATE、DELETE 和 SELECT FOR UPDATE。在事务的提交阶段 SELECT FOR UPDATE 读到的行，也会进行两阶段提交，因此，它们也可以参与事务冲突检测。如发生写入冲突，那么包含 SELECT FOR UPDATE 语句的事务会提交失败。如果没有冲突，事务将成功提交，当提交结束时，这些被加锁的行，会产生一个新版本，可以让其他尚未提交的事务，在将来提交时发现写入冲突。|
+|`FOR UPDATE` | 对查询结果集所有行上锁（对于在查询条件内，但是不在结果集的行，将不会加锁，如事务启动后由其他事务写入的行），以监测其他事务对这些的并发修改。使用[乐观事务模型](/v3.1/reference/transactions/transaction-optimistic.md)时，语句执行期间不会检测锁，因此，不会像 PostgreSQL 之类的数据库一样，在当前事务结束前阻止其他事务执行 UPDATE、DELETE 和 SELECT FOR UPDATE。在事务的提交阶段 SELECT FOR UPDATE 读到的行，也会进行两阶段提交，因此，它们也可以参与事务冲突检测。如发生写入冲突，那么包含 SELECT FOR UPDATE 语句的事务会提交失败。如果没有冲突，事务将成功提交，当提交结束时，这些被加锁的行，会产生一个新版本，可以让其他尚未提交的事务，在将来提交时发现写入冲突。若使用悲观事务，则行为与其他数据库基本相同，不一致之处参考[和 MySQL InnoDB 的差异](/v3.1/reference/transactions/transaction-pessimistic.md#和-mysql-innodb的差异)。|
```
Comment (lilin90) [inline v3.1/reference/sql/statements/select.md]:
```suggestion
|`FOR UPDATE` | 对查询结果集所有行上锁（对于在查询条件内，但是不在结果集的行，将不会加锁，如事务启动后由其他事务写入的行），以监测其他事务对这些的并发修改。使用[乐观事务模型](/v3.1/reference/transactions/transaction-optimistic.md)时，语句执行期间不会检测锁，因此，不会像 PostgreSQL 之类的数据库一样，在当前事务结束前阻止其他事务执行 `UPDATE`、`DELETE` 和 `SELECT FOR UPDATE`。在事务的提交阶段 `SELECT FOR UPDATE` 读到的行，也会进行两阶段提交，因此，它们也可以参与事务冲突检测。如发生写入冲突，那么包含 `SELECT FOR UPDATE` 语句的事务会提交失败。如果没有冲突，事务将成功提交，当提交结束时，这些被加锁的行，会产生一个新版本，可以让其他尚未提交的事务，在将来提交时发现写入冲突。若使用悲观事务，则行为与其他数据库基本相同，不一致之处参考[和 MySQL InnoDB 的差异](/v3.1/reference/transactions/transaction-pessimistic.md#和-mysql-innodb-的差异)。|
```

**Comment by anotherrachel** [comment]
All versions updated. PTAL, thanks! @lilin90 @youjiali1995  @sticnarf

File: v2.1/reference/transactions/overview.md [hunk 21]
```diff
@@ -159,4 +158,57 @@ insert into test values (3);
...
+COMMIT;
+```
+
+同理，执行 `INSERT` 语句时，建议使用显式事务。
+
+> **注意：**
+>
+> 由于 TiDB 中的资源是分布式的，TiDB 中单线程 workload 可能不会很好地利用分布式资源，因此性能相比于单实例部署的 MySQL 较低。这与 TiDB 中的事务延迟较高的情況类似。
+
+### 大事务
```
Comment (lilin90) [inline v2.1/reference/transactions/overview.md]: @youjiali1995 确认一下，TiDB v2.1 有大事务吗，是否只在某个版本开始有呢？

**Comment by TomShawn** [comment]
@anotherrachel Please update this PR according to https://github.com/pingcap/docs-cn/pull/2255, thanks!

File: v2.1/reference/transactions/overview.md (hunk 21 above)
Comment (youjiali1995) [reply v2.1/reference/transactions/overview.md]: 不好意思，刚看到。4.0 才开始有的

Comment (anotherrachel) [reply v2.1/reference/transactions/overview.md]: This confuses me... The doc that introduces transaction model (now deleted in this PR) used to say "大事务“, even in v2.1. See [here](https://github.com/jackysp/docs-cn/blob/3748fc3f297ba16458dc7d04f6b4151ab33706e1/v2.1/reference/transactions/transaction-model.md). 
So "大事务“ cannot be used here in v2.1 v3.0 and v3.1? @youjiali1995

Comment (sticnarf) [reply v2.1/reference/transactions/overview.md]: @anotherrachel The documentation are actually telling the user that TiDB has poor support of big transactions so there are some limitations.

Comment (sticnarf) [reply v2.1/reference/transactions/overview.md]: The limitations are relaxed in 4.0 so we say big transactions are supported in 4.0

Comment (anotherrachel) [reply v2.1/reference/transactions/overview.md]: @sticnarf Thank you for clarifying that! Then we can keep the current description about "大事务“ in v2.1/v3.0/v3.1? 
In dev version, I have already updated the description. See [here](https://github.com/pingcap/docs-cn/pull/2198/commits/d55f8f4c42e1afea711495733e845ee2ad21144e).

Comment (sticnarf) [reply v2.1/reference/transactions/overview.md]: SGTM

**Comment by anotherrachel** [comment]
@lilin90 PTAL, thanks!

**Pull request merged**